### DISCUSSION TITLE
Using two editors classic and extended type

### DISCUSSION THREAD

**Discussion by Kavakuo** [TypeFox/monaco-languageclient#740] [Q&A]
Hi all, I have a question regarding the usage of multiple editors. My use case is to have two editors. One editor is connected to a language server (extended config), the second editor is used to edit and validate JSON according to a schema using the default monaco JSON web worker (classic config). When I test both editors independently, everything works as expected. When both editors are visible at the same time or loaded after each other, it basically works but not perfectly (the syntax highlighting in the classic editor is broken).

<details>
<summary>Exception thrown when typing in the classic/json editor</summary>

![CleanShot 2024-08-27 at 13 22 02](https://github.com/user-attachments/assets/f8cf3a95-c24d-43f3-8484-f7118df7e64f)

</details>

I assume that this might be related to imports that are triggered for the extended editor ([here](https://github.com/TypeFox/monaco-languageclient/blob/main/packages/wrapper/src/editorAppExtended.ts#L72-L73) that trigger additional imports [here](https://github.com/CodinGame/monaco-vscode-api/blob/main/src/service-override/theme.ts#L10)).

Since I couldn't find anything related to this problem, I just wanted to ask if this is a known problem and if you might have an idea for a workaround.

**Comment by kaisalmen** [comment]
Hi @Kavakuo the service configurations are globally applied. It is either textmate or monarch. Both is not possible. You can have more than one editor, but they must be of the same type (classic/extended). Currently this is not checked. It should be possible to provide a proper warning to the user (bugfix/enhancement).
